lib/auth/auth_with_roles.go

```diff
@@ -3045,6 +3045,12 @@ func (a *ServerWithRoles) DeleteMFADevice(ctx context.Context) (proto.AuthServic
 	return nil, trace.NotImplemented("bug: DeleteMFADevice must not be called on auth.ServerWithRoles")
 }
 
+// DeleteMFADeviceSync is implemented by AuthService.DeleteMFADeviceSync.
+func (a *ServerWithRoles) DeleteMFADeviceSync(ctx context.Context, req *proto.DeleteMFADeviceSyncRequest) error {
+	// The token provides its own authorization and authentication.
```

Comment:
Mention that it is authorized by either token or context user?

Reply:
The delete request actually requires token in both cases (recovery token and later a privilege token).

Reply:
Right, nvm.